### DISCUSSION TITLE
Basic questions regarding dense search

### DISCUSSION THREAD

**Discussion by kormoczi** [castorini/pyserini#927] [Q&A] [answered]
Hi,
I could use sparse search based on the examples (both with prebuild indexes and with my own, custom made indexes as well).
But with dense search I have some problems.
I started to use the wikipedia-dpr-single-nq-bf (Faiss) index. But which encoder should I use? Is there any recommendation for this?
I have tried facebook/dpr-question_encoder-multiset-base and facebook/dpr-question_encoder-single-nq-base, and both looks ok (at least I can get some docid as search result), but do not know if there is any difference, which one is better, or shall I use something else?
And the even bigger problem is, that if the dense search gives the result, I cannot fetch the content with the searcher.doc() method, as I just receive an error, that: "AttributeError: 'SimpleDenseSearcher' object has no attribute 'ssearcher'"
Because this wikipedia index is very big, I assumed that the content is in it. Is it true or not?
If yes, how can I get access to the documents?
If not, which sparse index shell I use to get the document (as there are multiple ones for wikipedia), which one is compatible with this index (I mean the same docid gives same content)? All of them is OK, or shall I choose a specific one?
Thank you for any help or suggestion!

**Comment by kormoczi** [comment, marked as answer]
It looks like, that the wikipedia-dpr index has different docid convention, so I have to use the enwiki-paragraphs index to fetch the documents, after a search on the wikipedia-dpr-single-nq-bf index...

**Reply by lintool** [reply]
Seems like you answered your own question... :)

But yes, only the Lucene indexes (with `-storeRaw` index option) will store the text in the index.

The other question of what dense model to use: if you have a test collection, use the model that provides better results... if you don't have a test collection, there's really no clear guidance at present - it's really a question of cross-domain generalization, which is an active area of research - see https://arxiv.org/abs/2104.08663
